.github/workflows/swift-toolchain.yml

```diff
@@ -1728,84 +1639,113 @@ jobs:
+            $NdkFlags = ""
+
+            $StdlibCc = cygpath -m "${{ github.workspace }}/BinaryCache/Library/Developer/Toolchains/unknown-Asserts-development.xctoolchain/usr/bin/clang-cl.exe"
+            $StdlibCxx = $StdlibCc
+            $StdlibFlags = "-D MSVC_C_ARCHITECTURE_ID=${{ matrix.arch }} -D MSVC_CXX_ARCHITECTURE_ID=${{ matrix.arch }}"
+            $StdlibCacheFile = "${{ github.workspace }}/SourceCache/swift/cmake/caches/Runtime-${{ matrix.platform }}-${{ matrix.cpu }}.cmake"
+          } elseif ("${{ matrix.os }}" -eq "Android") {
+            $SwiftC = cygpath -m "${{ github.workspace }}/BinaryCache/Library/Developer/Toolchains/unknown-Asserts-development.xctoolchain/usr/bin/swiftc.exe"
+
+            $NDKPATH = cygpath -m "${{ steps.setup-ndk.outputs.ndk-path }}"
+            $SwiftArch = if ("${{ matrix.cpu }}" -eq "armv7") { "armv7a" } else { "${{ matrix.cpu }}" }
+            $CmakeCpu = if ("${{ matrix.cpu }}" -eq "armv7") { "armv7-a" } else { "${{ matrix.cpu }}" }
+
+            $SwiftFlags += " -sdk ${NDKPATH}/toolchains/llvm/prebuilt/windows-x86_64/sysroot"
+            $SwiftFlags += " -Xclang-linker -target ${SwiftArch}-unknown-linux-androideabi${{ inputs.ANDROID_API_LEVEL }}"
+            $SwiftFlags += " -Xclang-linker --sysroot -Xclang-linker $NDKPATH/toolchains/llvm/prebuilt/windows-x86_64/sysroot"
+            $SwiftFlags += " -Xclang-linker -resource-dir -Xclang-linker $NDKPATH/toolchains/llvm/prebuilt/windows-x86_64/lib/clang/17"
+
+            $InstallDir = cygpath -m "${{ github.workspace }}/BuildRoot/Library/Developer/Platforms/${{ matrix.os }}.platform/Developer/SDKs/${{ matrix.os }}.sdk/usr"
+            $NdkFlags = "-D CMAKE_ANDROID_NDK=`"${NDKPATH}`" -D SWIFT_ANDROID_NDK_PATH=`"${NDKPATH}`""
+
+            $StdlibCc = "${{ matrix.cc }}"
+            $StdlibCxx = "${{ matrix.cxx }}"
+            $StdlibFlags = "-D SWIFT_BUILD_CLANG_OVERLAYS_SKIP_BUILTIN_FLOAT=YES -D LLVM_ENABLE_LIBCXX=YES -D SWIFT_USE_LINKER=lld -D CMAKE_SYSTEM_PROCESSOR=${CmakeCpu}"
+            $StdlibCacheFile = "${{ github.workspace }}/SourceCache/swift/cmake/caches/Runtime-${{ matrix.platform }}-${{ matrix.cpu }}.cmake"
+          } else {
+            $SwiftC = "${{ github.workspace }}/BinaryCache/Library/Developer/Toolchains/unknown-Asserts-development.xctoolchain/usr/bin/swiftc"
+            $InstallDir = "${{ github.workspace }}/BuildRoot/Library/Developer/Toolchains/unknown-Asserts-development.xctoolchain/usr"
+            $NdkFlags = ""
+
+            $StdlibCc = "${{ github.workspace }}/BinaryCache/Library/Developer/Toolchains/unknown-Asserts-development.xctoolchain/usr/bin/clang"
+            $StdlibCxx = "${{ github.workspace }}/BinaryCache/Library/Developer/Toolchains/unknown-Asserts-development.xctoolchain/usr/bin/clang++"
+            $StdlibFlags = ""
+            $StdlibCacheFile = "${{ github.workspace }}/SourceCache/swift-build/Runtime-${{ matrix.platform }}-${{ matrix.arch }}.cmake"
+          }
```

Comment:
Can we push this down into the workflow through the matrix and simplify this somehow?

Reply:
I moved these to the matrices. It's a bit simpler now.